2019/d10Asteroids.d

```diff
@@ -0,0 +1,291 @@
+import std.array, std.algorithm, std.typecons;
+
+struct Pt { int x, y; int opCmp(ref const Pt s) const { return x == s.x ? y - s.y : x - s.x; } }
+
+struct NMV {
```

Comment:
Generally, it would be excellent practice to comment this code as it isn't obvious what's going on. Pretty please?